apps/binding.go

```diff
@@ -1,39 +1,128 @@
 package apps
 
+// Binding is the main way for an App to attach its functionality to the
```

Comment:
"main way" - might be helpful to add the others (with links) - maybe when Justine polishes this doc?

Reply:
cc @justinegeffen

Reply:
@larkox Binding is the _only_ way, isn't it? (I'm fuzzy on in-post, but it is a binding too, right?)

@larkox would be awesome to have a simple, standalone in-post "example" app?

Reply:
The only functionality that is not properly a binding is a form, but a form can only be opened by a binding.

Reply:
An example post (to be submitted through the post api) would be something like:
```
{
    channel_id: "channelID",
    message: "Some message to appear before the attachment",
    props: {
        app_bindings: [{
            app_id: "my app id",
            location: "location",
            label: "title of the attachment",
            description: "body of the attachment",
            bindings: [
                {
                    location: "my_select",
                    label: "Placeholder text",
                    bindings: [
                        {
                            location: "option1",
                            label: "Option 1",
                        }, {
                            location: "option2",
                            label: "Option 2",
                        },
                    ],
                    call: {
                        path: "my/path",
                    },
                }, {
                    location: "my_button",
                    label: "Button label",
                    call: {
                        path: "my/path",
                    },
                },
            ],
        }],
    },
}
```